core/app/status/logger.go

```diff
@@ -34,19 +35,26 @@ func RegisterLogger(progressUpdateFreq time.Duration) Unregister {
 }
 
 type statusLogger struct {
-	lastProgressUpdate map[*Task]time.Time
-	progressUpdateFreq time.Duration
+	lastProgressUpdate     map[*Task]time.Time
+	lastProgressUpdateLock sync.Mutex
+	progressUpdateFreq     time.Duration
 }
 
 func (l statusLogger) OnTaskStart(ctx context.Context, t *Task) {
 	log.I(ctx, "%v Started", t)
```

Comment:
Should this move inside the lock? Has a task really started if it's in a queue waiting to acquire a lock?

Reply:
Moved the log message to after the lock. Don't want to log while holding the lock.